tests/framework/nfs-utils.go

```diff
@@ -29,7 +29,11 @@ func createNFSPVs(client *kubernetes.Clientset, cdiNs string) error {
 
 func deleteNFSPVs(client *kubernetes.Clientset, cdiNs string) error {
 	for i := 1; i <= pvCount; i++ {
-		if err := utils.DeletePV(client, nfsPVDef(strconv.Itoa(i), utils.NfsService.Spec.ClusterIP)); err != nil {
+		pv := nfsPVDef(strconv.Itoa(i), utils.NfsService.Spec.ClusterIP)
+		if err := utils.DeletePV(client, pv); err != nil {
+			return err
+		}
+		if err := utils.WaitTimeoutForPVDeleted(client, pv, timeout); err != nil {
```

Comment:
Optimization here, right now you are doing the following:
1. Delete PV
2. Wait For PV to actually be gone
3. Go to 1
So you are sequentially waiting for the PV to be deleted before starting a new delete. How about you do
```go
for i := 1; i <= pvCount; i++ {
  pv := nfsPVDef(strconv.Itoa(i), utils.NfsService.Spec.ClusterIP)
  if err := utils.DeletePV(client, pv); err != nil {
    return err
  }
}
// kicked off deletion of all PVs I care about
for i := 1; i <= pvCount; i++ {
  if err := utils.WaitTimeoutForPVDeleted(client, pv, timeout); err != nil {
    return err
  }
}
```
In this order you start the deletion of the PVs before waiting for any of them to really be deleted. That way the deletion is in parallell while the waiting is sequential. And hopefully the wait will be less because when you get to the wait for the later PVs, they will be gone already.

Reply:
Good idea. done

Reply:
So the way it is now, it looks like you are waiting for just the last PV (since the PV variable is set to the last at the end of the first loop). But there is no guarantee in the order in which the PVs are deleted, it is possible at this point that some of the earlier ones aren't gone yet. I would either just re-create the pv defs in the second loop, or store them in a slice and loop over the slice in the second loop to ensure all the PVs are gone at the end.